From: Nick Piggin <npiggin@suse.de>
References: FATE309111
Subject: uninline add_to_page_cache
Patch-mainline: no

Uninline add_to_page_cache so that we can add to the function in a
subsequent patch (pagecache limiting) without changing the kABI.

Signed-off-by: Nick Piggin <npiggin@suse.de>
--
Index: linux-2.6.27/include/linux/pagemap.h
===================================================================
--- linux-2.6.27.orig/include/linux/pagemap.h
+++ linux-2.6.27/include/linux/pagemap.h
@@ -432,6 +432,8 @@ static inline int fault_in_pages_readabl
 	return ret;
 }
 
+int add_to_page_cache(struct page *page, struct address_space *mapping,
+				pgoff_t index, gfp_t gfp_mask);
 int add_to_page_cache_locked(struct page *page, struct address_space *mapping,
 				pgoff_t index, gfp_t gfp_mask);
 int add_to_page_cache_lru(struct page *page, struct address_space *mapping,
@@ -439,20 +441,4 @@ int add_to_page_cache_lru(struct page *p
 extern void remove_from_page_cache(struct page *page);
 extern void __remove_from_page_cache(struct page *page);
 
-/*
- * Like add_to_page_cache_locked, but used to add newly allocated pages:
- * the page is new, so we can just run __set_page_locked() against it.
- */
-static inline int add_to_page_cache(struct page *page,
-		struct address_space *mapping, pgoff_t offset, gfp_t gfp_mask)
-{
-	int error;
-
-	__set_page_locked(page);
-	error = add_to_page_cache_locked(page, mapping, offset, gfp_mask);
-	if (unlikely(error))
-		__clear_page_locked(page);
-	return error;
-}
-
 #endif /* _LINUX_PAGEMAP_H */
Index: linux-2.6.27/mm/filemap.c
===================================================================
--- linux-2.6.27.orig/mm/filemap.c
+++ linux-2.6.27/mm/filemap.c
@@ -521,6 +521,23 @@ out:
 }
 EXPORT_SYMBOL(add_to_page_cache_locked);
 
+/*
+ * Like add_to_page_cache_locked, but used to add newly allocated pages:
+ * the page is new, so we can just run __set_page_locked() against it.
+ */
+int add_to_page_cache(struct page *page,
+		struct address_space *mapping, pgoff_t offset, gfp_t gfp_mask)
+{
+	int error;
+
+	__set_page_locked(page);
+	error = add_to_page_cache_locked(page, mapping, offset, gfp_mask);
+	if (unlikely(error))
+		__clear_page_locked(page);
+	return error;
+}
+EXPORT_SYMBOL(add_to_page_cache);
+
 int add_to_page_cache_lru(struct page *page, struct address_space *mapping,
 				pgoff_t offset, gfp_t gfp_mask)
 {
